src/toGamut.js

```diff
@@ -104,3 +113,95 @@ export default function toGamut (color, {method = defaults.gamut_mapping, space
 }
 
 toGamut.returns = "color";
+
+/**
+ * Given a color `origin`, returns a new color that is in gamut using
+ * the CSS Gamut Mapping Algorithm. If `space` is specified, it will be in gamut
+ * in `space`, and returned in `space`. Otherwise, it will be in gamut and
+ * returned in the color space of `origin`.
+ * @param {Object} origin
+ * @param {Object} options
+ * @param {ColorSpace|string} options.space
+ * @returns {Color}
+ */
+export function toGamutCSS (origin, { space = origin.space }) {
+	const JND = 0.02;
+	const ε = 0.0001;
+	space = ColorSpace.get(space);
+
+	if (space.isUnbounded) {
+		return to(origin, space);
+	}
+
+	const origin_OKLCH = to(origin, ColorSpace.get("oklch"));
+	let L = origin_OKLCH.coords[0];
+
+	// return media white or black, if lightness is out of range
+	if (L >= 1) {
+		const white = to(parse("white"), space);
```

Comment:
I would advise against using `parse()` here. First because it's a slower code path than creating objects (and gamut mapping needs to be *very* fast) and second because the module that resolves keywords may not even be included (Color.js needs to be tree-shakable).

I would suggest just defining an object (outside of the function, since it doesn't change) in one of the color spaces that are already imported by the dependencies of this module (lab?).

Reply:
I implemented this in https://github.com/LeaVerou/color.js/pull/344/commits/118e6f752b4de00fed1485d30399a0579fe78377, choosing Oklab. It's used in the DeltaOK calculation, so the Oklab space is explicitly required for the algorithm.